### ISSUE TITLE
A failed to start OpenShift deployment marks as a successful build on Travis

### ISSUE THREAD

**Issue by mrsimonemms** [travis-ci/travis-ci#2956]
I got my NodeJS project to deploy from Travis to OpenShift.  There was an error in my `npm start -d` script which meant it wouldn't start (my fault, I'm ok with that).  What's most concerning though is that Travis marked this as a successful build - I would suggest that a failed deployment should fail the build.

The build URL is https://travis-ci.org/riggerthegeek/kitbag-api/builds/40939572.  Below is the relevant information, from line 1,439 onwards:

```
remote: Thu Nov 13 2014 17:29:58 GMT-0500 (EST): Starting application 'kitbagapi' ...
remote: *** NodeJS supervisor is disabled due to .openshift/markers/use_npm
remote: *** Starting application using: npm start -d
remote: Application 'kitbagapi' failed to start 1
remote: -------------------------
remote: Git Post-Receive Result: success
remote: Activation status: success
remote: Deployment completed with status: success
To ssh://54621084e0b8cdbd6a00014e@kitbagapi-riggerthegeek.rhcloud.com/~/git/kitbagapi.git/
a951f36..28f09e9 master -> master
Already up-to-date!
# HEAD detached at 28f09e9
# Untracked files:
# (use "git add <file>..." to include in what will be committed)
#
# .dpl/
#
nothing added to commit but untracked files present (use "git add" to track)
Dropped refs/stash@{0} (18f002183d7bba440f1b2992bfb10cfd3d90bb82)
Done. Your build exited with 0.
```

**Comment by caiocutrim** [comment]
How can i resolve this same problem? I don't understand...

**Comment by BanzaiMan** [comment]
@caiocutrim Are you seeing this problem now? https://github.com/travis-ci/dpl/commit/b876da8f90be883d5ccf13cda4301ceacbbc6d83 should allow `rhc` gem to be updated to the latest version, and the things should work automatically.

If you are having a problem, could you point to the build log that shows this problem?

**Comment by caiocutrim** [comment]
how can i update rhc?

**Comment by caiocutrim** [comment]
I can't deploy....
Warning: Permanently added 'contactooh-maranhao.rhcloud.com' (RSA) to the list of known hosts.Counting objects: 3573, done.Delta compression using up to 32 threads.Compressing objects: 100% (2636/2636), done.Writing objects: 100% (3573/3573), 5.08 MiB | 5.27 MiB/s, done.Total 3573 (delta 564), reused 3553 (delta 555)error: Could not read 18e918d91ed099b2921208befdc81bea63dde42cfatal: revision walk setup failederror: Could not read 18e918d91ed099b2921208befdc81bea63dde42cfatal: revision walk setup failedremote: Stopping NodeJS cartridgeremote: MongoDB already stoppedremote: Repairing links for 1 deploymentsTo ssh://54d935204382ec143a00007e@contactooh-maranhao.rhcloud.com/~/git/contactooh.git/ ! [remote rejected] master -> master (missing necessary objects)error: failed to push some refs to 'ssh://54d935204382ec143a00007e@contactooh-maranhao.rhcloud.com/~/git/contactooh.git/'Already up-to-date!# On branch master# Your branch is ahead of 'origin/master' by 1 commit.#   (use "git push" to publish your local commits)## Untracked files:#   (use "git add <file>..." to include in what will be committed)##	.dpl/#nothing added to commit but untracked files present (use "git add" to track)Dropped refs/stash@{0} (5c056d9ca5ee5901c4127e359954b58e413978e7)Done. Your build exited with 0.

**Comment by BanzaiMan** [comment]
@caiocutrim Could you reformat the log message? I can't quite figure out what is going on there. You could also point to the log message we can see.

**Comment by caiocutrim** [comment]
oh, i'm sorry...
https://travis-ci.org/caiocutrim/contactooh/builds/50142506

**Comment by BanzaiMan** [comment]
@caiocutrim No worries. Does your build create artifacts? If so, you should add `skip_cleanup: true`. (See http://docs.travis-ci.com/user/deployment/openshift/#Deploying-build-artifacts)

**Comment by caiocutrim** [comment]
It works! Thank you @BenzaiMan
https://travis-ci.org/caiocutrim/contactooh/builds/50150071

**Comment by caiocutrim** [comment]
now i have this...
https://travis-ci.org/caiocutrim/contactooh/builds/50153011#L2372
Warning: Permanently added 'contactooh-maranhao.rhcloud.com' (RSA) to the list of known hosts.Counting objects: 3452, done.Delta compression using up to 32 threads.Compressing objects: 100% (2582/2582), done.Writing objects: 100% (3452/3452), 4.83 MiB | 8.52 MiB/s, done.Total 3452 (delta 503), reused 3427 (delta 494)
error: Could not read 89310f60364d9e568682cd54411278adc36793e9fatal: revision walk setup failed 
error: Could not read 89310f60364d9e568682cd54411278adc36793e9
fatal: revision walk setup failedremote: Stopping NodeJS cartridgeremote: MongoDB already stopped 
remote: Repairing links for 1 deploymentsTo ssh://54d96a9ffcf933b79c000039@contactooh-maranhao.rhcloud.com/~/git/contactooh.git/ ! [remote rejected] master -> master (missing necessary objects)
error: failed to push some refs to 'ssh://54d96a9ffcf933b79c000039@contactooh-maranhao.rhcloud.com/~/git/contactooh.git/'

**Comment by travi** [comment]
I've been seeing this same issue. The latest build with the problem can be seen here: https://travis-ci.org/travi/travi-api/jobs/63428337

**Comment by Demven** [comment]
Hello everyone.
I've found a solution!
Just add line to travis yml to skip cleanup (this will save your changes and new files after your build process, if you use gulp, grunt or any other similar build tools)

deploy:
...
    skip_cleanup: true

!!! Then remove your .gitignore file (it causes travis to ignore those new files created by gulp to commit them to OpenShift).
I created special gulp task to remove it only during travis deploy

var del = require('del');
...
gulp.task('travis:remove:gitignore', function (cb) {
  return del(['.gitignore'], cb);
});
...
gulp.task('production', function (callback) {
  runSequence('clean', 'build', 'travis:remove:gitignore', callback);
});
